misk/src/main/kotlin/misk/web/actions/WebActionMetadata.kt

```diff
@@ -0,0 +1,51 @@
+package misk.web.actions
+
+import misk.ApplicationInterceptor
+import misk.web.DispatchMechanism
+import misk.web.NetworkInterceptor
+import misk.web.PathPattern
+import misk.web.mediatype.MediaRange
+import okhttp3.MediaType
+import kotlin.reflect.KType
+
+data class WebActionMetadata(
+  val name: String,
+  val function: String,
+  val functionAnnotations: List<String>,
+  val requestMediaTypes: List<String> = listOf(),
+  val responseMediaType: String?,
+  val parameterTypes: List<String> = listOf(),
+  val returnType: String,
+  val pathPattern: String,
+  val interceptorsApplication: List<String> = listOf(),
+  val interceptorsNetwork: List<String> = listOf(),
+  val dispatchMechanism: DispatchMechanism
+)
+
+internal fun WebActionMetadata(
+  name: String,
+  function: Function<*>,
+  functionAnnotations: List<Annotation> = listOf(),
```

Comment:
worth having defaults for these? Are they ever used?

Reply:
`name` will always be used and can always be deduced by the action name which is a required field
`function` same for function
`functionAnnotations` might be used to discern roles/services but can be empty so `listOf()` seems to be a useful default

Reply:
jk, I'll remove all the defaults